crates/pallets/starknet/src/runtime_api.rs

```diff
@@ -62,6 +63,8 @@ sp_api::decl_runtime_apis! {
         fn get_tx_execution_outcome(tx_hash: TransactionHash) -> Option<Vec<u8>>;
         /// Return messages sent to L1 during tx execution
         fn get_tx_messages_to_l1(tx_hash: TransactionHash) -> Vec<MessageToL1>;
+        /// Return two lists of messages sent (to L1) and consumed (from L1) during this block
+        fn get_starknet_messages(block_extrinsics: Vec<<Block as BlockT>::Extrinsic>, chain_id: Felt252Wrapper) -> (Vec<MessageL2ToL1>, Vec<MessageL1ToL2>);
```

Comment:
What dictates the signature return type? What will it be used by

Reply:
Starknet os output struct has two fields with the respectful types, I added a code snippet to illustrate that (pr description)